### PR TITLE
Enhance speed benchmark for storages

### PR DESCRIPTION
**Pull request by not522**
## Motivation
This is a part of #4500.

The current speed benchmark CI does not measure the performance of storages enough.

## Description of the changes
- Add benchmark for `RDBStorage` with `TPESampler` and `CmaEsSampler`
- Add benchmark for `JournalStorage`

### CODE REVIEW

File: benchmarks/asv/optimize.py [hunk 1]
```diff
@@ -63,7 +63,11 @@ def time_optimize(self, args: str) -> None:
         "inmemory, tpe, 1000",
         "inmemory, cmaes, 1000",
         "sqlite, random, 1000",
-        "cached_sqlite, random, 1000",
```
Comment (not522) [inline benchmarks/asv/optimize.py:66]: I removed it because `sqlite` and `cached_sqlite` are equivalent in the current implementation.

**Comment by not522** [comment]
@contramundum53 @gen740 Could you review this PR?

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Review by gen740 — APPROVED** [review]
LGTM!

**Comment by gen740** [comment]
@contramundum53 Something is wrong with CI. Could you rerun it?

**Comment by not522** [comment]
Hmm, it looks like the concurrency is doing something wrong. I close this PR and send another PR.

**Pull request closed without merging**